tools/package-source.js

```diff
@@ -99,6 +99,93 @@ var splitConstraint = function (c) {
            constraint: parsed.constraintString || null };
 };
 
+// Given the text of a README.md file, excerpts the text between the first and
+// second heading.
+//
+// Specifically - if there is text between the document name, and the first
+// subheading, it will take that text. If there is no text there, and only text
+// after the first subheading, it will take that text. It won't look any deeper
+// than that (in case the user intentionally wants to leave the section blank
+// for some reason). Skips lines that start with an exclamation point.
+var getExcerptFromReadme = function (text) {
+  // Split into lines.
+  var fullText = text.split("\n");
+  if (! fullText) {
+    return "";
+  }
+
+  // The lines of the excerpt that we will return to the caller.
+  var excerpt = [];
+
+  // We want to take the data between the first and second headers, so it is
+  // important to keep track of how many headers we have passed.
+  var headerNum = 0;
+
+  // Counter and termination condition for the while loop.
+  var lineNum = 0;
+  var done = false;
+
+  // We want to scan until we have an excerpt, or decide that we are not getting
+  // one. If we are only taking the first five lines of a long README file, we
+  // want to avoid scanning all of it.
+  while (! done) {
+    var currentLine = fullText[lineNum];
+    var trimmedLine = currentLine.trim();
+
+    // According to "https://help.github.com/articles/markdown-basics/", a
+    // heading is denoted by a '#' symbol, though, I think, the line at the
+    // top (===) should also count as a header for our purposes.
+    var isHeader =
+      (trimmedLine[0] === "#") ||
+      (trimmedLine[0] === "=");
```

Comment:
I will probably poke `commonmark` a bit more today -- I guess the bug that I found yesterday isn't *terrible* and I like `commonmark` in general. Just wanted to throw the above stuff out there, as a thought.